Cabal/src/Distribution/Simple/Command.hs

```diff
@@ -588,11 +591,13 @@ data CommandParse flags
   | CommandList [String]
   | CommandErrors [String]
   | CommandReadyToGo flags
+  | CommandDelegate
 instance Functor CommandParse where
   fmap _ (CommandHelp help) = CommandHelp help
   fmap _ (CommandList opts) = CommandList opts
   fmap _ (CommandErrors errs) = CommandErrors errs
   fmap f (CommandReadyToGo flags) = CommandReadyToGo (f flags)
+  fmap _ _ = CommandDelegate
```

Comment:
Catch-alls can be dangerous if you add constructors in the future (you will not be warned by `-Wall` on incomplete pattern matching). Can you add a relevant constructor here?

Reply:
Indeed, catch-alls are dangerous, well-spotted!